app/javascript/topNavigation/utilities.js

```diff
@@ -1,3 +1,106 @@
+function closeHeaderMenu(memberMenu, menuNavButton) {
+  if (menuNavButton) {
+    menuNavButton.setAttribute('aria-expanded', 'false');
+  }
+
+  if (memberMenu) {
+    setTimeout(() => {
+      memberMenu.classList.remove('showing');
+    }, 5);
+  }
+}
+
+function blurHeaderMenu(memberMenu, menuNavButton, potentiallyActiveElement) {
+  setTimeout(() => {
+    if (document.activeElement !== potentiallyActiveElement) {
+      closeHeaderMenu(memberMenu, menuNavButton);
+    }
+  }, 10);
+}
+
+function toggleHeaderMenu(memberMenu, navigationButton) {
+  if (!memberMenu || !navigationButton) {
+    return;
+  }
+
+  let crayonsHeaderMenuClassList = memberMenu.classList;
+  if (crayonsHeaderMenuClassList.contains('showing')) {
+    crayonsHeaderMenuClassList.remove('showing');
+    navigationButton.setAttribute('aria-expanded', 'false');
+  } else {
+    crayonsHeaderMenuClassList.add('showing');
+    navigationButton.setAttribute('aria-expanded', 'true');
+
+    let firstNavLink = document.getElementById('first-nav-link');
+    if (firstNavLink) {
+      setTimeout(() => {
+        // focus first item on open
+        firstNavLink.focus();
+      }, 100);
+    }
+  }
+}
+
+export function isTouchDevice() {
+  return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini|DEV-Native-ios/i.test(
+    navigator.userAgent,
+  );
+}
+
+export function initializeTouchDevice(memberTopMenu, menuNavButton) {
+  if (navigator.userAgent === 'DEV-Native-ios') {
+    document.body.classList.add('dev-ios-native-body');
+  }
+
+  if (memberTopMenu) {
+    let crayonsHeaderMenuClassList = memberTopMenu.classList;
+
+    setTimeout(() => {
+      closeHeaderMenu(memberTopMenu, menuNavButton);
+
+      if (isTouchDevice()) {
+        // Use a named function instead of anonymous so duplicate event handlers are discarded
+        menuNavButton.addEventListener('click', function (_event) {
+          toggleHeaderMenu(memberTopMenu, menuNavButton);
+        });
+      } else {
+        crayonsHeaderMenuClassList.add('desktop');
+        menuNavButton.addEventListener('click', function (_event) {
+          toggleHeaderMenu(memberTopMenu, menuNavButton);
+        });
+        memberTopMenu.addEventListener('keyup', (e) => {
+          if (
+            e.key === 'Escape' &&
+            crayonsHeaderMenuClassList.contains('showing')
+          ) {
+            crayonsHeaderMenuClassList.remove('showing');
+            menuNavButton.focus();
+          }
+        });
+        document
+          .getElementById('last-nav-link')
+          .addEventListener('blur', function (_event) {
+            blurHeaderMenu(
+              memberTopMenu,
+              menuNavButton,
+              document.getElementById('seond-last-nav-link'),
```

Comment:
This looks to be a typo with the `seond-last-nav-link` [sic]. Even if corrected, does that Id match any element on the page? I wasn't actually sure.

Reply:
I'll take a peek. Thanks @marcysutton!

Reply:
I've added `second-last-nav-link` to the appropriate menu element and corrected the typo so now we don't close the menu if a user tabs backwards from the last option 👍